Use fonts from the Pattern Library

andy-armstrong opened the pull request:
## [FEDX-56](https://openedx.atlassian.net/browse/FEDX-56)
- Removed checked-in versions of the OpenSans fonts
- Changes the Sass to use the fonts from the Pattern Library
### Sandbox
- [ ] https://andy-armstrong.sandbox.edx.org/
### Testing
- [ ] Manual testing of the LMS and Studio (including TinyMCE which updated)
### Reviewers

If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @bjacobel 
- [x] Code review: @AlasdairSwan 

FYI: @dan-f @cahrens 
### Post-review
- [x] Squash commits

edx-webhook commented:
# PR Cover Letter
- [ ] A detailed description of the changes in the body of the PR
- [ ] If available, a link to the relevant JIRA ticket in the body of the PR
  - `[PROJ-XXXX](https://openedx.atlassian.net/browse/PROJ-XXXX)`

# Sandbox
- [ ] Build a sandbox for your branch and add a link here

# Testing
- [ ] i18n
- [ ] RTL
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Analytics
- [ ] Performance
- [ ] Database migrations are backwards-compatible

# Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review #1
- [ ] Code review #2
- [ ] UI strings/error msgs review
- [ ] UX review
- [ ] Accessibility review
- [ ] Product review

# DevOps assistance
If it's needed, or even if you suspect it might be needed, do any/all these things as early as possible. Don't surprise devops right before merging! For example, touching envs/aws.py or or envs/common.py are dead giveaways that you'll need assistance when pushing your changes out. Things that require devops attention include, but are not limited to:
- configuration changes
- feature flags
- migrations
Ways to get devops attention, pick-and-choose whichever is most appropriate for your specific PR:
- [ ] Tag devops on your PR (`@edx/devops`), either for review (more specific to devops = better) or just a general FYI
- [ ] File a support ticket

FYI: Tag anyone who might be interested in this PR here.

# Post-review
- [ ] Squash commits

andy-armstrong commented:
@clrux Maybe a good first step on the Pattern Library front would be to get this story merged.

andy-armstrong commented:
@bjacobel Can you take this PR over as part of your work on completing the accessible icons.

bjacobel commented:
Sure, what's the work remaining on it?

andy-armstrong commented:
@bjacobel I think probably just manual verification on a sandbox, plus any thing you and @clrux agree on to make the Pattern Library fonts and icons consumable in the platform.

You probably should look at @clrux's work in the following PR too: https://github.com/edx/edx-platform/pull/11438

andy-armstrong commented:
@bjacobel I've simplified this PR by removing the edx-icons changes, so now this just focuses on the OpenSans fonts. We have a separate story to remove edx-icons from the platform so it doesn't make sense to mess with that here.

andy-armstrong commented:
@robrap Can you review the second commit here, which fixes safe-commit-linter.sh to not attempt to lint files that were deleted in a commit.

robrap commented:
@andy-armstrong: I'm guessing it failed due to an outdated safe-commit-linter.sh file, and the latest should work as is. Do you mind verifying before making any changes? Thanks.

andy-armstrong commented:
@robrap I rebased yesterday so I don't think that's the case. To be certain I just rebased again this morning and pushed up this PR without my linter changes. When I run it locally it fails with the same error that I was seeing yesterday. As far as I can see, safe-commit-linter.sh finds every file from the commit and tries to run safe_template_linter.py against it. This includes files that were deleted in the commit. When it tries to run the linter against the deleted file it gives the error I was seeing.

robrap commented:
@andy-armstrong: Thanks. I'm not clear on what is going wrong. Verbose mode on the script shows the commits used and the diff command. The supposedly fixed git diff filters on copies, adds, and modifies. I'll take a look when I get in.

bjacobel commented:
Diff and stager LGTM.

andy-armstrong commented:
@bjacobel @AlasdairSwan Please re-review this change that I've finally resurrected now that I know how  to fix the screenshot Bok Choy test failure. This PR just removes all the OpenSans fonts from edx-platform and instead uses them from the npm installed copy of the Pattern Library.

AlasdairSwan commented:
jenkins run bokchoy

AlasdairSwan commented:
This looks good, but I was unaware of the checked in CSS file ```common/static/css/tinymce-studio-content-fonts.css```. I assume it is out of scope for this PR but it would be great to move that to being compiled by Sass in the future if possible. 👍

bjacobel commented:
:thumbsup:

andy-armstrong commented:
jenkins run bokchoy

bjacobel commented:
I think this should be closed, @alisan617's work supercedes this.

andy-armstrong commented:
@bjacobel I don't believe that @alisan617 did anything with the OpenSans fonts, so I believe this is still relevant. In particular, the files I'm deleting here have not been deleted yet.

I've rebased and I'll see if I can get this back on track and then merged.

andy-armstrong commented:
jenkins run bokchoy

andy-armstrong commented:
jenkins run bokchoy

andy-armstrong commented:
jenkins run lettuce

andy-armstrong commented:
jenkins run bokchoy

andy-armstrong commented:
jenkins run bokchoy

andy-armstrong commented:
@muhammad-ammar I'm trying to merge this PR, but it is breaking two of the focus tests that you added in this PR: https://github.com/edx/edx-platform/commit/c3b6b10478393ba26608bed60d36e998b6fbb540

I know that we've had a lot of trouble with focus tests in the past, so I'm somewhat tempted to leave them disabled. Do you have any insight as to why they might fail? Are they known to be flaky? I've disabled them for now in my PR, so let me know if you're okay with this, or if we should try to get them to work?

andy-armstrong commented:
@muhammad-ammar one other point: I don't know how changing fonts could cause focus tests to fail. I couldn't find anything that seemed to depend upon exact pixel offsets, but maybe there is something you're aware of.

muhammad-ammar commented:
@andy-armstrong I haven't added any focus tests in above mentioned commit. :) In fact there no changes made by me in the mentioned commit. :) My name is coming due to some issue with git history re-write. The whole work is done by Cambridge team. You may need to have a discussion with @cahrens

andy-armstrong commented:
Thanks @muhammad-ammar. That's messed up that GitHub is giving out the wrong history! @cahrens, can you help me with these failing focus tests?

cahrens commented on common/test/acceptance/tests/lms/test_problem_types.py:
```diff
@@ -167,7 +167,8 @@ def test_answer_correctly(self):
         self.problem_page.click_review_in_notification()
-        self.assertTrue(self.problem_page.is_focus_on_problem_meta())
+        # TODO: determine why the focus is not being set
+        # self.assertTrue(self.problem_page.is_focus_on_problem_meta())
```
We have an existing flaky bok choy ticket (https://openedx.atlassian.net/browse/TNL-5774) for calling is_focus_on_problem_meta in a different test point within this class (test_reset_clears_answer_and_focus). I'll add information to TNL-5774 to include investigating and re-enabling these usages too.

Qubad786 commented on common/test/acceptance/tests/video/test_video_module.py:
```diff
@@ -1223,7 +1224,7 @@ def test_if_captions_are_draggable(self):
         else:
             self.assertEqual(
-                captions_end.get('y') + 15,
+                captions_end.get('y') + 16,
```
Hey @andy-armstrong what was the context behind this change? I am getting bokchoy failure on my PR #15247 which would get pass if we do `captions_end.get('y') + 15` – Failure message says: 689.0 != 688.0 Closed captions did not get dragged.

andy-armstrong replied:
@Qubad786 the context was I got the same kind of failure, and so I changed the number! It seems that these drag-and-drop tests are very pixel sensitive. My assumption was that somehow the fonts were slightly different, which caused the drag targets to be slightly off. I don't see anything in your code that would change how these targets would render, but if they reliably pass with the updated number then by all means change it back.
